src/arithmetic/numeric_funcs/numeric_funcs.c

```diff
@@ -135,6 +135,16 @@ SIValue AR_SQRT(SIValue *argv, int argc) {
 	return SI_DoubleVal(sqrt(SI_GET_NUMERIC(arg)));
 }
 
+// returns base^exponent
+SIValue AR_POW(SIValue *argv, int argc) {
+	SIValue base = argv[0];
+	SIValue exp = argv[1];
+	// return NULL if input is none numeric
+	if(!_validate_numeric(base) || !_validate_numeric(exp)) return SI_NullVal();
```

Comment:
Arithmetic expression validation will make sure `base` and `exp` are either numeric or NULL.
```c
if (SIValue_IsNull(base) || SIValue_IsNull(exp)) return SI_NullVal();
```

Reply:
I don't think what you are saying is correct:
static inline int _validate_numeric(const SIValue v) {
	return SI_TYPE(v) & SI_NUMERIC;
}

#define SI_TYPE(value) (value).type
#define SI_NUMERIC (T_INT64 | T_DOUBLE)

Reply:
When this function (POW) is registered, we specify that this function accepts 2 arguments, first one is of type NUMERIC or NULL, second one is of type NUMERIC or NULL

here we want to check only if either one is NULL and if so we need to return NULL

Reply:
So this is what this function does:
If any of the arguments is invalid or NULL it returns NULL.

Reply:
It is cheaper to just test for NULL. you're already guarantee that `base` and `exp` are either numeric or  NULL